### DISCUSSION TITLE
How to inject Csla classes into code-behind files

### DISCUSSION THREAD

**Discussion by ecaldentey** [MarimerLLC/csla#4103] [Questions] [answered]
Hello Rocky,

In my Razor pages, I am using code-behind to separate the Razor and HTML code from the C# code, but the Csla classes cannot be injected into the code-behind file and must be injected into the Razor file to be used in the code-behind file.

Am I doing something wrong, or is it definitely not possible to inject Csla classes directly into the code-behind file?

How can I inject Csla classes into code-behind files?

Thanks

**Comment by StefanOssendorf** [comment, marked as answer]
Hi. You should be able to inject e.g. `IDataPortal<T>` like you would inject any other c# class into your code behind file.
I'm don't know how razor pages work but it sounds like you can inject other classes into them.

**Comment by ecaldentey** [comment]
@StefanOssendorf 

You're absolutely right, I apologize for my mistake, thank you for the clarification.
